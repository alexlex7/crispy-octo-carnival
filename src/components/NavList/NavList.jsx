import React from 'react';
import styles from './NavList.module.css';
import NavListItem from '../NavListItem/NavListItem';
import PropTypes from 'prop-types';

export default function NavList({ pages }) {
  return (
    <nav>
      <ul className={styles.list}>
        {pages.map(({ text, to }) => (
          <NavListItem key={text} text={text} to={to} />
        ))}
      </ul>
    </nav>
  );
}

NavList.propTypes = {
  pages: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      to: PropTypes.string.isRequired,
    })
  ),
};
