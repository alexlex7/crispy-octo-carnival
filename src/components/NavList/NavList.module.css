@media screen and (max-width: 1204px) {
  .list {
    display: flex;
    flex-direction: column;
    align-items: center;
    align-content: center;
    row-gap: 20px;
  }
}

@media screen and (min-width: 1205px) {
  .list {
    display: flex;
    column-gap: 50px;
    list-style: none;
  }
}
