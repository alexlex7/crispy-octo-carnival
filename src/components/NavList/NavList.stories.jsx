import React from 'react';
import NavList from './NavList';

export const pages = [
  { text: 'Home', to: '/' },
  { text: 'About', to: 'about' },
  { text: 'Features', to: 'features' },
  { text: 'Contact', to: 'contact' },
];

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Navigation/NavList',
  component: NavList,
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <NavList {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  pages,
};
