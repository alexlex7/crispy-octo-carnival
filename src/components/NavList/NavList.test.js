import { render, screen } from '@testing-library/react';
import NavList from './NavList';
import { BrowserRouter } from 'react-router-dom';

const pages = [
  { text: 'Home', to: '/' },
  { text: 'About', to: 'about' },
  { text: 'Features', to: 'features' },
  { text: 'Contact', to: 'contact' },
];

test('renders navlist component', () => {
  const { asFragment } = render(<NavList pages={pages} />, {
    wrapper: BrowserRouter,
  });
  const linkElement = screen.getByRole('list');
  expect(linkElement).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
