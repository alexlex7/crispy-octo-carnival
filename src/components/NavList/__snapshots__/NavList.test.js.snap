// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navlist component 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="list"
    >
      <li
        class="listItem"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="item"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="item"
          href="/features"
        >
          Features
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="item"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
